.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,131 @@
+---
+name: Build and Publish
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [esp32, esp8266]
+        appliance:
+          - name: electric_tank_water_heater
+            shorthand: etwh
+          - name: heatpump_water_heater
+            shorthand: hpwh
+          - name: hvac
+            shorthand: hvac
+          - name: tankless_water_heater
+            shorthand: tlwh
+      fail-fast: false
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4.1.1
+      - name: Build manifest
+        uses: mikefarah/yq@v4.35.2
+        with:
+          cmd: yq -P '.packages.econet = "econet_${{ matrix.appliance.name }}.yaml" | .packages.econet   tag = "!include"' ./build-yaml/${{ matrix.platform }}.yaml > econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}.yaml
+      - name: Build firmware
+        uses: esphome/build-action@v1.8.0
+        id: esphome-build
+        with:
+          yaml_file: econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}.yaml
+          version: latest
+      - name: Copy firmware and manifest
+        run: |
+          mkdir output
+          mv ${{ steps.esphome-build.outputs.name }} output/
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3.1.3
+        with:
+          name: econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}
+          path: output
+          retention-days: 7
+
+  consolidate-manifests:
+    # if: github.event_name != 'pull_request' && github.ref == 'refs/heads/main'
+    name: Consolidate ${{ matrix.appliance.shorthand }} manifests
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        appliance:
+          - name: electric_tank_water_heater
+            shorthand: etwh
+          - name: heatpump_water_heater
+            shorthand: hpwh
+          - name: hvac
+            shorthand: hvac
+          - name: tankless_water_heater
+            shorthand: tlwh
+      fail-fast: false
+    steps:
+      - name: Download built firmwares
+        uses: actions/download-artifact@v3.0.2
+        with:
+          path: firmwares
+      - name: Dump Files
+        run: |-
+          ls -R firmwares
+      - name: Copy files
```

Comment:
This doesn't seem to copy any files

Reply:
Yes, I added that to make debugging much simpler as otherwise it is very hard to tell what files are generated and how laid out in directory structure. I'm inclined to leave it for future troubleshooting as the cost is basically 0.